src/threading.c

```diff
@@ -408,16 +408,24 @@ jl_ptls_t jl_init_threadtls(int16_t tid)
 
 JL_DLLEXPORT jl_gcframe_t **jl_adopt_thread(void) JL_NOTSAFEPOINT_LEAVE
 {
+    // jl_init_threadtls puts us in a GC unsafe region, so ensure GC isn't running.
+    // we can't use a normal safepoint because we don't have signal handlers yet.
+    // we also can't use jl_safepoint_wait_gc because that assumes we're in a task.
+    jl_atomic_fetch_add(&jl_gc_disable_counter, 1);
```

Comment:
and this release?

Reply:
`jl_atomic_fetch_add` is implemented using `std::atomic_fetch_add`, and AFAICT with no order specified that uses seq_cst, so should be fine. And our `_explicit` versions take a C++ ordering, so I can't use those from `threading.c`.